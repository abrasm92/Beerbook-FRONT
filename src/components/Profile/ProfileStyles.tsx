import styled from "styled-components";

const ProfileStyles = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  .profile {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    list-style: none;
    height: 100%;
    min-width: 280px;
    width: 80%;
    max-width: 400px;
    overflow: hidden;
    border: none;
    padding: 0 20px 20px 20px;

    p {
      font-size: 1rem;
      padding-bottom: 10px;
    }

    button {
      background-color: #0d47a1;
      border: none;
      border-radius: 7px;
      color: #fff;
      padding: 7px;
      cursor: pointer;
    }

    .profile--main-info {
      display: flex;
      margin-bottom: 15px;
      h2 {
        font-size: 1rem;
        margin-bottom: 10px;
      }
      img {
        margin-right: 30px;
        width: 100px;
        height: 100px;
        object-fit: contain;
        border-radius: 20px;
      }
    }

    .profile--edit {
      font-weight: 700;
      font-size: 0.9rem;
      margin-bottom: 20px;
    }

    .profile--lists {
      margin-top: 30px;
      padding-bottom: 20px;
      width: 230px;
      h3 {
        font-size: 1rem;
      }
      border-bottom: 1px solid #b6b6b6;
    }

    .profile--lists * {
      margin-top: 10px;
    }

    .profile--button-add-beer {
      font-size: larger;
      margin: 0 auto;
      margin-top: 140px;
      margin-bottom: 20px;
    }
  }
`;

export default ProfileStyles;
