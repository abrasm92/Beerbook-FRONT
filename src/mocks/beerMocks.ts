import { BeerDataApi } from "../types/interfaces";

export const singleBeer: BeerDataApi = {
  id: "6297a8369ab4b03687f15b72",
  name: "Sofie",
  brewery: "Goose Island",
  style: "Belgian season",
  degrees: 6.5,
  IBU: 25,
  country: "EEUU",
  description:
    "Cerveza de color ámbar, con aromas intensos de cítricos, con final lig…",
  image: "",
  imageBackup: "",
  owner: "admin",
};

export const groupOfBeer: BeerDataApi[] = [
  {
    id: "6297a8369ab4b03687f15b72",
    name: "Sofie",
    brewery: "Goose Island",
    style: "Belgian season",
    degrees: 6.5,
    IBU: 25,
    country: "EEUU",
    description:
      "Cerveza de color ámbar, con aromas intensos de cítricos, con final lig…",
    image: "",
    imageBackup: "",
    owner: "",
  },
  {
    id: "6297a8369ab4b03687f15b73",
    name: "Breakfast Stout",
    brewery: "Founders",
    style: "Imperial Stout",
    degrees: 8.3,
    IBU: 60,
    country: "EEUU",
    description:
      "Cerveza negra con gran espuma color marrón con arómas predominantes a …",
    image: "",
    imageBackup: "",
    owner: "",
  },
  {
    id: "6297a8369ab4b03687f15b7a",
    name: "Ginette Natural Christmas",
    brewery: "Brasserie La Binchoise",
    style: "Belgian Strong Ale",
    degrees: 8.5,
    IBU: NaN,
    country: "Bélgica",
    description:
      "Cerveza rojiza oscura con espuma densa, cuerpo medio suave y aromatica…",
    image: "",
    imageBackup: "",
    owner: "",
  },
];
