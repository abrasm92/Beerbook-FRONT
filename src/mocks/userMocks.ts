import { BasicUser, LoginUser, User, UserData } from "../types/interfaces";

export const singleUser: User = {
  name: "admin",
  username: "admin",
  email: "admin@admin.com",
  password: "beerbook",
};

export const loginUser: LoginUser = {
  username: "admin",
  password: "beerbook",
};

export const userLogged: BasicUser = {
  id: "1234",
  name: "admin",
};

export const userData: UserData = {
  name: "admin",
  username: "admin",
  email: "admin@admin.com",
  image: "",
  imageBackup: "",
  creations: 10,
  favorites: 23,
  age: 29,
  country: "espa√±a",
  admin: true,
};
