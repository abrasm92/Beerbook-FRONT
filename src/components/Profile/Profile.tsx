import { UserData } from "../../types/interfaces";
import ProfileStyles from "./ProfileStyles";

type ProfileProp = {
  user: UserData;
};

const Profile = ({ user }: ProfileProp): JSX.Element => {
  return (
    <ProfileStyles>
      <div className="profile">
        <div className="profile--main-info">
          <img
            src={
              user.imageBackup
                ? user.imageBackup
                : "/images/default-profile-image.jpg"
            }
            alt="imagen de perfil"
          />
          <div className="profile--main-info-text">
            <h2>{user.name}</h2>
            <p>
              {user.age ? `${user.age} años` : ""}{" "}
              {user.age && user.country ? "·" : ""}{" "}
              {user.country ? user.country : ""}
            </p>
          </div>
        </div>
        <p className="profile--edit">Editar perfil</p>
        <div className="profile--lists">
          <h3>Total aportaciones - {user.creations}</h3>
          <button>Ver aportaciones</button>
        </div>
        <div className="profile--lists">
          <h3>Mis favoritas - {user.favorites}</h3>
          <button>Ver favoritas</button>
        </div>
        <button className="profile--button-add-beer">
          Agregar una cerveza
        </button>
      </div>
    </ProfileStyles>
  );
};

export default Profile;
