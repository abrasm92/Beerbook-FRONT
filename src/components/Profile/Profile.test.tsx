import { userData } from "../../mocks/userMocks";
import Profile from "./Profile";
import reactTestRenderer from "react-test-renderer";

describe("Given a Profile componet", () => {
  describe("When it's rendered", () => {
    test("Then it should show always the same as snapshot", () => {
      const profile = reactTestRenderer.create(<Profile user={userData} />);

      expect(profile).toMatchSnapshot();
    });
  });
});
