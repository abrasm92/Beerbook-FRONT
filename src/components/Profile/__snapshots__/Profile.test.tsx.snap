// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Profile componet When it's rendered Then it should show always the same as snapshot 1`] = `
<section
  className="sc-bczRLJ eyYaNv"
>
  <div
    className="profile"
  >
    <div
      className="profile--main-info"
    >
      <img
        alt="imagen de perfil"
        src="/images/default-profile-image.jpg"
      />
      <div
        className="profile--main-info-text"
      >
        <h2>
          admin
        </h2>
        <p>
          29 años
           
          ·
           
          españa
        </p>
      </div>
    </div>
    <p
      className="profile--edit"
    >
      Editar perfil
    </p>
    <div
      className="profile--lists"
    >
      <h3>
        Total aportaciones - 
        10
      </h3>
      <button>
        Ver aportaciones
      </button>
    </div>
    <div
      className="profile--lists"
    >
      <h3>
        Mis favoritas - 
        23
      </h3>
      <button>
        Ver favoritas
      </button>
    </div>
  </div>
</section>
`;
